builtin/src/lib.rs

```diff
@@ -21,7 +22,10 @@ pub fn setup() -> ExecutionWrangler {
     )
     .unwrap();
     let spec = GenesisSpec::default(manifest_data_cid);
-    let _genesis = create_genesis_actors(&mut builder, &spec).unwrap();
-    let bench = builder.build().unwrap();
+    let genesis = create_genesis_actors(&mut builder, &spec).unwrap();
+    // check that the genesis state matches assumptions in the builtin-actors test code
+    assert_eq!(genesis.faucet_id, TEST_FAUCET_ADDR.id().unwrap());
+    let circulating_supply = spec.reward_balance + spec.faucet_balance;
```

Comment:
The reward actor balance isn't part of the circulating supply. Does everything work ok if you remove that here?

Reply:
I think perhaps the name `circulating_supply` is inappropriate for the purpose here.

In builtin-actors the implementation for the TestVM also includes reward balance. https://github.com/filecoin-project/builtin-actors/blob/3eb3cec1213f5e5a64ca848069f70ca15db392e7/test_vm/src/lib.rs#L96

It's then used here to inject an expectation into the check_state_invariants. https://github.com/filecoin-project/builtin-actors/blob/3eb3cec1213f5e5a64ca848069f70ca15db392e7/integration_tests/src/util/mod.rs#L61. This check needs to compare an expectation with the sum of all actors' balances.

Reply:
The existing code around circulating supply is generally sloppy and wrong. From tests point of view, it's generally just an input needed for some calculations.

The invariant check you linked to should be using something called total supply instead, which is the sum of all balances (and is something the VM should enforce, not the actors). So maybe you can call this total supply, but it's then not the right thing for the circulating supply syscall to return for calculations, and we need such a value too. But I guess you can leave that wrong for now.